modules/nf-neuro/preproc/n4/meta.yml

```diff
@@ -1,7 +1,36 @@
 ---
 # yaml-language-server: $schema=https://raw.githubusercontent.com/scilus/nf-neuro/main/modules/meta-schema.json
 name: "preproc_n4"
-description: Bias field correction using N4
+description: |
+  Perform N4 bias field correction on either an anatomical image or a diffusion
+  volume. For diffusion volumes, the module automatically determines the
+  bspline parameter, number of stages, and iteration pyramid based on the image
+  resolution and size.
+
+  The formula used are as follows:
+  For the BSpline parameter:
+  `BSplineParam = 2 ** (NbStages - 1) * BSplineKnotPerVoxel * ShrinkFactor * ImageResolution`
+
+  For the number of stages in the iteration pyramid:
+  If SmallestDimension ≤ BSplineKnotPerVoxel * ShrinkFactor:
+      NbStages = 1
+  Else:
+      NbStages = ceil( log2(SmallestDimension / (BSplineKnotPerVoxel * ShrinkFactor)) )
+
+  For the iteration pyramid:
+  Slope     = (MinIter - MaxIter) / (1 - Retain)
+  Intercept = MaxIter - Slope * Retain
+
+  For i in linspace(0, 1, NbStages):
+      If i < Retain:
+          Iteration = MaxIter
+      Else:
+          Iteration = round(Slope * i + Intercept)
```

Comment:
Here a variable is called BSplineKnotPerVoxel but the code is nb_voxels_between_knots

I think nb_voxels_between_knots is probably the correct concept?